chapter/background.md

```diff
@@ -0,0 +1,607 @@
+Although this example targets AWS, Nix itself is platform-agnostic and NixOps supports different backends through various plugins.
+Configurations like this are abstractions over many manual steps and the Nix language employed in this example allows for even higher level turing-complete interaction with configurations.
+
+```{.nix #lst:nixops-rproxy caption="Example NixOps deployment to AWS"}
+{
+  network.description = "Gollum server and reverse proxy";
+  defaults = 
+    { config, pkgs, ... }:
+    {
+      # Configuration of a specific deployment implementation
+      # here: AWS EC2
+
+      deployment.targetEnv = "ec2";
+      deployment.ec2.accessKeyId = "AKIA...";
+      deployment.ec2.keyPair = "...";
+      deployment.ec2.privateKey = "...";
+      deployment.ec2.securityGroups = pkgs.lib.mkDefault [ "default" ];
+      deployment.ec2.region = pkgs.lib.mkDefault "eu-west-1";
+      deployment.ec2.instanceType = pkgs.lib.mkDefault "t2.large";
+    };
+  gollum =
+    { config, pkgs, ... }:
+    {
+      # Nix based setup of the gollum server
+
+      services.gollum = {
+        enable = true;
+        port = 40273;
+      };
+      networking.firewall.allowedTCPPorts = [ config.services.gollum.port ];
+    };
+
+  reverseproxy =
+    { config, pkgs, nodes, ... }:
+    let
+      gollumPort = nodes.gollum.config.services.gollum.port;
+    in
+    {
+      # Nix based setup of a nginx reverse proxy
+
+      services.nginx = {
+        enable = true;
+        virtualHosts."wiki.example.net".locations."/" = {
+          proxyPass = "http://gollum:${toString gollumPort}";
+        };
+      };
+      networking.firewall.allowedTCPPorts = [ 80 ];
+      
+      # Instance can override default deployment options 
+      deployment.ec2.instanceType = "t1.medium";
+    };
+}
+```
+
+Similarly, tools like Terraform[@terraform], or Chef[@chef] use their own DSLs and integrate with most major cloud providers.
+The popularity of these products^[https://trends.google.com/trends/explore?date=2012-01-01%202022-01-01&q=%2Fg%2F11g6bg27fp,CloudFormation], beyond all, highlights the importance of expressive configuration formats and their industry value.
+
+Finally, descriptive data formats for cloud configurations allow mitigating security risks through static analysis.
+Yet, as recently as spring 2020 and still more than a year later dossiers of Palo Alto Networks' security department Unit 42 [@pa2020H1, ps2021H2] show that a majority of public projects uses insecure configurations.
+This suggests that techniques[@aws-cloud-formation-security-tests] to automatically check templates are not actively employed, and points out the importance of evaluated configuration languages which can implement passive approaches to security analysis.
+
+### Nickel
+
+The Nickel[@nickel] language is a configuration programming language (cf. [@sec:configuration-programming-languages]) with the aims of providing composable, verifiable and validatable configuration files.
```

Comment:
"configuration programming language" do you want to make a difference between "config language" and "configuration programming language"? if yes, first define both terms, and then say "Nickel is a configuration programming language because XXXX"

Reply:
```suggestion
The Nickel[@nickel] language is a configuration programming language as defined in  [@sec:configuration-programming-languages] with the aims of providing composable, verifiable and validatable configuration files.
```

configuration (programming) languages are defined previous to this chapter.
What makes Nickel a programming language is discussed in the following sentences